AWSAppSyncClient/Internal/AWSPerformOfflineMutationOperation.swift

```diff
@@ -64,22 +64,23 @@ final class AWSPerformOfflineMutationOperation: AsynchronousOperation, Cancellab
 
     private func notifyCompletion(_ result: JSONObject?, error: Error?) {
         operationCompletionBlock?(self, error)
+        withExtendedLifetime(self) {
+            handlerQueue.async { [weak self] in
```

Comment:
Can you explain how the change works? 
withExtendedLifetime will make sure that self will not destroyed before its closure return, but here handlerQueue.async  will return before handlerQueue.async's closure return, so withExtendedLifetime still cannot make sure the closure of handlerQueue.async  is finished before self is destroyed

Reply:
I was under the impression that the scope (self) would be maintained as long as anything was unfinished with "withExtendedLifetime".  Is that wrong?  (If so can you link me to an explanation of this, if possible?).

Maybe a simpler option would be to add a third argument to notifyCompletion, a callback function.  Then the state of the object can be switched to finished in there.  I can make the update and push today if you think that would make more sense

Reply:
Ok I did a toy example just to confirm you were right (which you are):
https://gist.github.com/tony7126/16de12ea7360ebda9344d3ec413d9ba5

Reply:
Here is a working toy example of the callback solution:
https://gist.github.com/tony7126/7160067f7d7bc48cedf68c4421744f1d

Let me know if you think something similar to that makes sense